src/methods/createAndSendTx.ts

```diff
@@ -51,8 +51,21 @@ export async function createAndSendTx(
     },
   };
   let options = txArgs.immortality ? { signer, era: 0 } : { signer };
-  await txExtrinsic.signAndSend(address, options);
-  // exit();
+
+  // Only resolve when it's finalised
+  await new Promise<void>((res) => {
+    txExtrinsic.signAndSend(address, options, ({ events = [], status }) => {
+      console.log("Transaction status:", status.type);
+
+      if (status.isInBlock) {
```

Comment:
We should add the case where the tx is refused or discarded

Reply:
I added handling but I don't know how to test it since it throws errors before sending the tx